src/LobbyServer.hs

```diff
@@ -224,26 +229,33 @@ changeNickName remoteClientList remoteGames newName = do
   mVarClientList <- remoteClientList
   clientList <- liftIO $ CC.readMVar mVarClientList
   sid <- getSessionID
```

Comment:
I know this function is a mess right now, the code will be better when the switch to using only the DB to store usernames is complete. Right now the username is saved both to the database and in the clientList the server holds.